README.md

```diff
@@ -57,6 +58,110 @@ class TreeContainer extends React.PureComponent {
 }
 ```
 
+## <a name="tree-structure"></a>Tree Structure
+
+A `tree` is an `object` with the following structure:
+
+```
+/**
+ * @type object
+ */
+Tree
+  /**
+   * The text being visualized.
+   * @type string
+   * @required
+   */
+  text: 'Sam likes eating hot dogs.'
+  /**
+   * The root node of the tree.
+   * @type object
+   * @required
+   */
+  root: Node { ... }
+```
+
+The `root` property refers to the root node of the tree to be visualized. Each `node` has the following
+structure:
+
+```
+/**
+ * @type object
+ */
+Node
+  /**
+   * The text content of the node
+   * @type string
+   * @required
+   *
+   * TODO: This will likely be migrated to be named `text` in a future version, as it's less specific.
+   */
+  word: 'eating'
+  /**
+   * A string specifying the "type" of node. This is used to determine it's color -- all nodes of
+   * the same type will be assigned the same color.
+   * @type string
+   * @required
+   */
+  nodeType: 'verb'
+  /**
+   * A string specifying describing the relationship between the node and it's parent. This text
+   * will be displayed on an element connecting the node and it's parent.
+   * @type string
+   * @optional
+   */
+  link: 'direct object'
+  /**
+   * An array of strings, which will be displayed on the node.
+   * @type string[]
+   * @optional
+   */
+  attributes: [ 'action', ... ]
+  /**
+   * An array of spans, where each span represents a series of characters in the `text` property (
+   * of the Tree) that should be highlighted when the node is hovered.
+   * @type object[]
+   * @optional
+   */
+  spans: [ Span { ... }, ... ]
+  /**
+   * An array containing the children of the node.
+   * @type object[]
+   * @optional
+   */
+  children: [ Node, ... ]
+```
+
+Each `span` refers to a sequence of characters in the original sentence (the `text` property of the
+`Tree`) that should be highlighted when the node and is hovered. Each `span` should have the
+following properties:
+
+```
+Span
+  /**
+   * The index indicating where the span begins.
+   * @type number
+   * @required
+   */
+  start
+  /**
+   * The index (exclusive) where the span ends.
+   * @type number
+   * @required
+   */
+  end
+  /**
+   * An optional identifier indicating the type of span. As of now, the only value you'll likely
+   * put here is "ignored", which indicates that the span shouldn't be emphasized when the node
+   * is hovered.
+   * @type string
+   * @optional
+   */
+  spanType
+```
```

Comment:
Can we add comments w/ example values for `spanType`? The documentation for this field seems light compared to the others.

Reply:
We could, however, the idea now is that the user shouldn't set it, unless they want to use the `ignored` value.  So it's effectively intentionally light, as documenting all of the span types seems unlikely.

I'll cut a ticket to blow out the `spanType` documentation so that it's accessible / consumable by users, but for now I think `ignored` or nothing is ok and keeps things simple -- does that make sense to you?